draft-gpew-priv-ppm.md

```diff
@@ -1442,35 +1442,62 @@ defined by the PPM parameters. (See {{batch-parameter-validation}}.)
-   correctly, a malicious client can skew the output of F only by reporting
-   false (untruthful) input. The output cannot be influenced in any other way.
-
-There are several additional constraints that a Prio deployment must satisfy in
-order to achieve these goals:
-
-1. Minimum batch size. The aggregation batch size has an obvious impact on
-   privacy. (A batch size of one hides nothing of the input.)
-2. Aggregation function choice. Some aggregation functions leak slightly more
-   than the function output itself.
-
-[TODO: discuss these in more detail.]
+PPM assumes an active attacker that controls the network and has the ability to
+statically corrupt any number of clients, aggregators, and collectors. That is,
+the attacker can learn the secret state of any party prior to the start of its
+attack. For example, it may coerce a client into providing malicious input
+shares for aggregation or coerce an aggregator into not diverting from the
+protocol specified (e.g., by divulging its input shares to the attacker).
+
+In the presence of this adversary, PPM aims to achieve the following high-level
+secure aggregation goals:
+
+1. Privacy. Clients trust that some aggregator is honest. That is, as long as at
+   least one aggregator executes the protocol faithfully as specified, and only
+   executes the protocol as specified, the parties learn nothing beyond the
+   aggregate result (i.e., the output of the aggregation function computed over
+   the honest measurements).
+1. Correctness. The collector trusts that the aggregators execute the protocol
+   correctly. That is, as long as the aggregators execute their roles as
+   specified, a malicious client can skew the aggregate result only by reporting
+   a false (untruthful) measurement. The result cannot be influenced in any
+   other way.
+
+To achieve these goals, there are a variety of attack vectors that a PPM
+deployment must consider:
+
+1. When crafted maliciously, collect requests may leak more information about
+   the measurements than the system intends. For example, the spec currently
+   allows sequences of collect requests to reveal an aggregate result for a
+   batch smaller than the minimum batch size. [OPEN ISSUE: See issue#195. This
+   also has implications for how we solve issue#183.]
+1. Even benign collect requests may leak information beyond what one might
+   expect intuitively. For example, the Poplar1 VDAF
+   {{?VDAF=I-D.draft-irtf-cfrg-vdaf}} can be used to compute the set of heavy
+   hitters among a set of arbitrary bit strings uploaded by clients. This
+   requires multiple evaluations of the VDAF, the results of which reveal
+   information to the aggregators and collector beyond what follows from the
+   heavy hitters themselves. Note that this leakage can be mitigated using
+   differential privacy. [OPEN ISSUE: We have yet not specified how to add DP.]
+1. The core PPM spec does not defend against Sybil attacks. In this type of
```

Comment:
Does "core" mean a deployment that doesn't use report extensions or any kind of client authentication mechanism?

Reply:
Beyond the protocol extensions it is also the case that some VDAFs have Sybil protections (against privacy) built-in e.g. with noise addition / differential privacy. PPM _could_ enforce that the aggregation function provide this kind of privacy but we choose not to.

Reply:
@tgeoghegan "core" refers to the bits of the spec that apply to all implementations. In particular, not every implementation will be required to interpret extensions the same way.

@csharrison It's not clear to me that DP always provides adequate protection against Sybil attacks. I think it ultimately depends on how you define privacy. That said, requiring DP is totally on the table, but before doing this I think we should start by spelling out what this would look like as an optional feature of the protocol. I think the first question to ask is whether this mechanism lives here or in the underlying VDAF. I would love to have your help on this.

Reply:
> @csharrison It's not clear to me that DP always provides adequate protection against Sybil attacks. I think it ultimately depends on how you define privacy. That said, requiring DP is totally on the table, but before doing this I think we should start by spelling out what this would look like as an optional feature of the protocol. I think the first question to ask is whether this mechanism lives here or in the underlying VDAF. I would love to have your help on this.

I think if you can prove that the system achieves user-level DP you should be able to show that even with worst-case Sybil attacks, a user's privacy is still protected to some degree. That being said, this is a property of the whole system, not just the VDAF.

I am fairly sure that DAP / PPM does not want to require DP in its deployments. This brings on a lot of baggage, and for instances where these kinds of attacks are truly not a problem (e.g. with airtight client authentication) it might not be that useful. The main point here is that we might want to call out that the particular VDAF used may mitigate some of the risks of Sybil.

Reply:
If achievable, I think specifying an optional, generic DP mechanism here (perhaps for some class of VDAFs with a specific property) would have significant advantages. IIUC, tuning and enforcing the privacy budget will require knowledge of how measurements are generated by clients over time. This is something that will be dictated by the deployment, and I think it would be best to avoid bleeding deployment details into VDAF wherever possible.

That said, VDAF may need to at least say something about how/when to add noise into an input or aggregate share.

Reply:
> If achievable, I think specifying an optional, generic DP mechanism here (perhaps for some class of VDAFs with a specific property) would have significant advantages

I think this is possible. We can probably have some tunable, exposed property of a VDAF that is a necessary but not sufficient condition for achieving DP. I think we can describe the rest of the conditions needed in the DAP deployment to get the rest of the way there.

I think this will not be super easy to do, but I am happy to help think through what it will take (maybe in a separate issue).

Reply:
Kicking off an issue is probably a good way forward. I'm happy to help with the protocol design, but will need to lean on you to make sure we have all the plumbing we need to implement. A couple seed questions to start:
1. Who adds noise and when? Do we want to support local DP, centralized DP, or both? Also would be helpful to define these (and point to references where needed).
2. What syntactic changes are necessary for VDAF?

Once you create the issue, it would also be helpful to ping the list with a brief description of the issue and a link.

Reply:
commented on https://github.com/ietf-wg-ppm/ppm-specification/issues/19, I doesn't answer all the questions but it's a start :)